### PR TITLE
feat(env): increase max readers limit depending on number of workers

### PR DESCRIPTION
**Pull request by Oyami-Srk**
[KAG-5700](https://konghq.atlassian.net/browse/KAG-5700)
https://github.com/Kong/kong/issues/13779

[KAG-5700]: https://konghq.atlassian.net/browse/KAG-5700?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dndx — CHANGES_REQUESTED** [review]
Let's write a test with very large `worker_processes` to confirm the fix.

**Comment by Oyami-Srk** [comment]
Test is added as `t/12-max-readers.t`

File: src/ngx_lua_resty_lmdb_module.c [hunk 1]
```diff
@@ -222,11 +225,30 @@ ngx_lua_resty_lmdb_open_file(ngx_cycle_t *cycle,
...
+    size_t                     readers;
+    ngx_core_conf_t            *ccf;
 
     if (ngx_lua_resty_lmdb_create_env(cycle, lcf, is_master) != NGX_OK) {
         return NGX_ERROR;
     }
 
+    /* Set max readers depending on the number of worker processes */
+    ccf = (ngx_core_conf_t *) ngx_get_conf(cycle->conf_ctx, ngx_core_module);
+    readers = (size_t)ccf->worker_processes;
```
Comment (dndx) [inline src/ngx_lua_resty_lmdb_module.c:237]:
```suggestion
    readers = (size_t) ccf->worker_processes;
```

File: src/ngx_lua_resty_lmdb_module.c [hunk 2]
```diff
@@ -222,11 +225,30 @@ ngx_lua_resty_lmdb_open_file(ngx_cycle_t *cycle,
...
+    readers += NGX_LUA_RESTY_LMDB_MAX_READERS_REDUNDANCY;
+    readers = readers > NGX_LUA_RESTY_LMDB_DEFAULT_READERS ?
+              readers : NGX_LUA_RESTY_LMDB_DEFAULT_READERS;
```
Comment (dndx) [inline src/ngx_lua_resty_lmdb_module.c:240]:
```suggestion
              readers : NGX_LUA_RESTY_LMDB_DEFAULT_READERS;

```

**Review by dndx — CHANGES_REQUESTED** [review]

File: src/ngx_lua_resty_lmdb_module.c [hunk 3]
```diff
@@ -222,11 +225,30 @@ ngx_lua_resty_lmdb_open_file(ngx_cycle_t *cycle,
 {
     int                        rc;
     int                        dead;
+    size_t                     readers;
+    ngx_core_conf_t            *ccf;
```
Comment (chronolaw) [inline src/ngx_lua_resty_lmdb_module.c:229]: style, should be:
```
  readers;
 *ccf;
```

File: src/ngx_lua_resty_lmdb_module.c [hunk 4]
```diff
@@ -222,11 +225,30 @@ ngx_lua_resty_lmdb_open_file(ngx_cycle_t *cycle,
...
 
     if (ngx_lua_resty_lmdb_create_env(cycle, lcf, is_master) != NGX_OK) {
         return NGX_ERROR;
     }
 
+    /* Set max readers depending on the number of worker processes */
+    ccf = (ngx_core_conf_t *) ngx_get_conf(cycle->conf_ctx, ngx_core_module);
+    readers = (size_t)ccf->worker_processes;
+    readers += NGX_LUA_RESTY_LMDB_MAX_READERS_REDUNDANCY;
+    readers = readers > NGX_LUA_RESTY_LMDB_DEFAULT_READERS ?
```
Comment (chronolaw) [inline src/ngx_lua_resty_lmdb_module.c:239]: `if ` may be easier to read?
```
if (readers < NGX_LUA_RESTY_LMDB_DEFAULT_READERS) {
  readers = NGX_LUA_RESTY_LMDB_DEFAULT_READERS
}
```

File: src/ngx_lua_resty_lmdb_module.c [hunk 5]
```diff
@@ -222,11 +225,30 @@ ngx_lua_resty_lmdb_open_file(ngx_cycle_t *cycle,
...
+    ngx_core_conf_t            *ccf;
 
     if (ngx_lua_resty_lmdb_create_env(cycle, lcf, is_master) != NGX_OK) {
         return NGX_ERROR;
     }
 
+    /* Set max readers depending on the number of worker processes */
+    ccf = (ngx_core_conf_t *) ngx_get_conf(cycle->conf_ctx, ngx_core_module);
+    readers = (size_t)ccf->worker_processes;
+    readers += NGX_LUA_RESTY_LMDB_MAX_READERS_REDUNDANCY;
```
[the remark this answers is not in the document]
Comment (Oyami-Srk) [reply src/ngx_lua_resty_lmdb_module.c]: done.

File: src/ngx_lua_resty_lmdb_module.c (hunk 3 above)
Comment (Oyami-Srk) [reply src/ngx_lua_resty_lmdb_module.c]: done.

File: src/ngx_lua_resty_lmdb_module.c (hunk 1 above)
Comment (Oyami-Srk) [reply src/ngx_lua_resty_lmdb_module.c]: done.

**Review by chronolaw — APPROVED** [review]

File: src/ngx_lua_resty_lmdb_module.c [hunk 6]
```diff
@@ -222,11 +225,31 @@ ngx_lua_resty_lmdb_open_file(ngx_cycle_t *cycle,
...
+    size_t                     readers;
+    ngx_core_conf_t           *ccf;
 
     if (ngx_lua_resty_lmdb_create_env(cycle, lcf, is_master) != NGX_OK) {
         return NGX_ERROR;
     }
 
+    /* Set max readers depending on the number of worker processes */
+    ccf = (ngx_core_conf_t *) ngx_get_conf(cycle->conf_ctx, ngx_core_module);
+    readers = (size_t) ccf->worker_processes + NGX_LUA_RESTY_LMDB_MAX_READERS_REDUNDANCY;
```
Comment (dndx) [inline src/ngx_lua_resty_lmdb_module.c:237]:
```suggestion
    readers = (size_t) ccf->worker_processes + NGX_LUA_RESTY_LMDB_MAX_READERS_REDUNDANCY;

```

**Review by dndx — APPROVED** [review]

**Pull request merged**